From 6f423c88f9f308f2e5b5494b3a01393030710a46 Mon Sep 17 00:00:00 2001
From: Jingchang Lu <jingchang.lu@freescale.com>
Date: Fri, 9 May 2014 02:28:49 +0800
Subject: [PATCH 086/255] arm: dts: ls1021a-qds: fix IFC nor and nand device
 node size

Signed-off-by: Jingchang Lu <jingchang.lu@freescale.com>
[Kevin: Original patch taken from
LS1021A-SDK-V1.1-ARM-SOURCE-20140815-yocto.iso]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm/boot/dts/ls1021a-qds.dts | 57 +++++++--------------------------------
 1 file changed, 9 insertions(+), 48 deletions(-)

diff --git a/arch/arm/boot/dts/ls1021a-qds.dts b/arch/arm/boot/dts/ls1021a-qds.dts
index b091217..febe972 100644
--- a/arch/arm/boot/dts/ls1021a-qds.dts
+++ b/arch/arm/boot/dts/ls1021a-qds.dts
@@ -195,48 +195,21 @@
 	status = "okay";
 	#address-cells = <2>;
 	#size-cells = <1>;
-	/* NOR, NAND Flashes and CPLD on board */
-	ranges = <0x1 0x0 0x68000000 0x08000000
+	/* NOR, NAND Flashes and FPGA on board */
+	ranges = <0x0 0x0 0x60000000 0x08000000
 		0x2 0x0 0x7e800000 0x00010000
 		0x3 0x0 0x7fb00000 0x00000100>;
 
-		nor@1 {
+		nor@0,0 {
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "cfi-flash";
 			reg = <0x1 0x0 0x8000000>;
 			bank-width = <2>;
 			device-width = <1>;
-
-			partition@40000 {
-				/* 256KB for DTB Image */
-				reg = <0x00040000 0x00040000>;
-				label = "NOR DTB Image";
-			};
-
-			partition@80000 {
-				/* 7 MB for Linux Kernel Image */
-				reg = <0x00080000 0x00700000>;
-				label = "NOR Linux Kernel Image";
-			};
-
-			partition@800000 {
-				/* 20MB for JFFS2 based Root file System */
-				reg = <0x00800000 0x01400000>;
-				label = "NOR JFFS2 Root File System";
-			};
-
-			partition@1f00000 {
-				/* This location must not be altered  */
-				/* 512KB for u-boot Bootloader Image */
-				/* 512KB for u-boot Environment Variables */
-				reg = <0x01f00000 0x00100000>;
-				label = "NOR U-Boot Image";
-				read-only;
-			};
 		};
 
-		nand@2 {
+		nand@2,0 {
 			#address-cells = <1>;
 			#size-cells = <1>;
 			compatible = "fsl,ifc-nand";
@@ -257,28 +230,16 @@
 			};
 
 			partition@200000 {
-				/* 4MB for Linux Kernel Image */
-				reg = <0x00200000 0x00400000>;
+				/* 10MB for Linux Kernel Image */
+				reg = <0x00200000 0x00a00000>;
 				label = "NAND Linux Kernel Image";
 			};
 
-			partition@600000 {
-				/* 4MB for Compressed Root file System Image */
-				reg = <0x00600000 0x00400000>;
+			partition@c00000 {
+				/* 500MB for Root file System Image */
+				reg = <0x00c00000 0x1f400000>;
 				label = "NAND Compressed RFS Image";
 			};
-
-			partition@a00000 {
-				/* 15MB for JFFS2 based Root file System */
-				reg = <0x00a00000 0x00f00000>;
-				label = "NAND JFFS2 Root File System";
-			};
-
-			partition@1900000 {
-				/* 7MB for User Area */
-				reg = <0x01900000 0x00700000>;
-				label = "NAND User area";
-			};
 		};
 
 		fpga: board-control@3,0 {
-- 
2.0.2

